File: database/src/main/java/it/niedermann/nextcloud/deck/database/dao/JoinBoardWithPermissionDao.java
package it.niedermann.nextcloud.deck.database.dao;

import androidx.room.Dao;
import androidx.room.Query;

import it.niedermann.nextcloud.deck.model.JoinBoardWithPermission;

@Dao
public interface JoinBoardWithPermissionDao extends GenericDao<JoinBoardWithPermission> {
    @Query("DELETE FROM joinboardwithpermission WHERE boardId = :localId")
    void deleteByBoardId(long localId);

}

File: data/local/src/main/java/it/niedermann/nextcloud/deck/data/local/dao/JoinBoardWithPermissionDao.kt
package it.niedermann.nextcloud.deck.data.local.dao

import androidx.room3.Dao
import androidx.room3.Query
import it.niedermann.nextcloud.deck.data.local.entity.JoinBoardWithPermissionEntity
import java.util.concurrent.CompletableFuture

@Dao
interface JoinBoardWithPermissionDao : GenericDao<JoinBoardWithPermissionEntity> {
    @Query("DELETE FROM JoinBoardWithPermission WHERE boardId = :boardId")
    fun deleteByBoardId(boardId: Long): CompletableFuture<Void?>
}
